src/python/pants/backend/jvm/targets/scala_library.py

```diff
@@ -82,3 +92,59 @@ def java_sources(self):
         raise TargetDefinitionException(self, 'No such java target: {}'.format(spec))
       targets.append(target)
     return targets
+
+  @property
+  def scalac_plugins(self):
+    """The names of compiler plugins to use when compiling this target with scalac.
+    :return: See constructor.
+    :rtype: list of strings.
+    """
+    if self._scoverage_instance.get_options().enable_scoverage:
+      # Prevent instrumenting generated targets and targets in blacklist.
+      if self.identifier.startswith(".pants.d.gen") or self._scoverage_instance.is_blacklisted(self):
```

Comment:
Rather than checking for `.pants.d`, you should use `if not self.is_synthetic`. As below, it would be good to do this in the target constructor though.

Reply:
Does doing this in the constructor means checking `kwargs['address'] in kwargs['build_graph'].synthetic_addresses`? I tried that, however, it is still instrumenting those targets.